code/controllers/subsystem/air.dm

```diff
@@ -284,14 +284,8 @@ SUBSYSTEM_DEF(air)
 			add_to_active(S)
 
 /datum/controller/subsystem/air/proc/setup_allturfs(list/turfs_to_init = block(locate(1, 1, 1), locate(world.maxx, world.maxy, world.maxz)))
-	var/list/active_turfs = src.active_turfs
-
-	// Clear active turfs - faster than removing every single turf in the world
-	// one-by-one, and Initalize_Atmos only ever adds `src` back in.
-	active_turfs.Cut()
-
-	for(var/thing in turfs_to_init)
-		var/turf/T = thing
+	for(var/turf/T in turfs_to_init)
```

Comment:
`block` can only return a list of of turfs, so this would be better to keep as a typeless list.

Reply:
i've tested three variations on 514.1549 and was unable to determine if any is faster than the other

```dm
/datum/controller/subsystem/air/proc/setup_allturfs(list/turfs_to_init = block(locate(1, 1, 1), locate(world.maxx, world.maxy, world.maxz)))
	for(var/turf/T in turfs_to_init)
		remove_from_active(T)
		if(T.blocks_air)
			continue
		T.Initialize_Atmos(times_fired)
		//CHECK_TICK

/datum/controller/subsystem/air/proc/setup_allturfs2(list/turfs_to_init = block(locate(1, 1, 1), locate(world.maxx, world.maxy, world.maxz)))
	for(var/T in turfs_to_init)
		remove_from_active(T)
		if(T:blocks_air)
			continue
		T:Initialize_Atmos(times_fired)
		//CHECK_TICK

/datum/controller/subsystem/air/proc/setup_allturfs3(list/turfs_to_init = block(locate(1, 1, 1), locate(world.maxx, world.maxy, world.maxz)))
	for(var/t in turfs_to_init)
		var/turf/T = t
		remove_from_active(T)
		if(T.blocks_air)
			continue
		T.Initialize_Atmos(times_fired)
		//CHECK_TICK
```

Reply:
Typed loops implicitly do an `istype` check on every single thing in their list, which adds to total CPU; for some things, it's going to be difficult to profile, but it's there.

If a loop is all of one particular type (or subtypes) and you intend to act on every single one, we go with typeless lists for performance reasons.

see: https://github.com/ParadiseSS13/Paradise/blob/master/.github/CONTRIBUTING.md istypeless loops.